test/functional/treesitter/parser_spec.lua

```diff
@@ -235,6 +235,41 @@ void ui_refresh(void)
     }, res)
   end)
 
+  it('can match special regex characters like \\ * + ( with `vim-match?`', function()
+    if pending_c_parser(pending) then return end
+
+    insert('char* astring = "\\n"; (1 + 1) * 2 != 2;')
+
+    local res = exec_lua([[
+      cquery = vim.treesitter.parse_query("c", '((_) @plus (vim-match? @plus "^\\\\+$"))'..
+                                               '((_) @times (vim-match? @times "^\\\\*$"))'..
+                                               '((_) @paren (vim-match? @paren "^\\\\($"))'..
+                                               '((_) @escape (vim-match? @escape "^\\\\\\\\n$"))'..
+                                               '((_) @string (vim-match? @string "^\\"\\\\\\\\n\\"$"))')
```

Comment:
`"^\\\\($"` would be `"^\\($"` in an actual query file. You cannot just write  `"^\($"` and then apply vim.escape because it would be invalid syntax in the query file. We cannot apply vim.escape before reading the file because then you could not escape `"` in strings with `\"`

Reply:
At the moment you can use `*` `+` via `[*]` or `[+]` but I don't know how to use `[` without escaping